### PR TITLE
[Android] Select the highest priority threat from SafetyNetClient.loo…

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
This PR changes the way how BraveSafeBrowsingApiHandler handles the case when SafetyNet reports more than one threat.
Before: the first one was taken.
Now: the most prioritized is taken, threats are arranged by priorities at `SAFETY_NET_THREAT_PRIORITIES` array.

`SAFETY_NET_THREAT_PRIORITIES` can be changed after the review, the test must be updated accordingly then.

The codes returned from https://testsafebrowsing.appspot.com/ are:

- (1) `[A/W/M/L/C] Should show a phishing warning` - 5 - `SafetyNetJavaThreatType.SOCIAL_ENGINEERING`
- (2) `[A/W/M/L/C/D] Should show a malware warning` - 4 - `SafetyNetJavaThreatType.POTENTIALLY_HARMFUL_APPLICATION`
- (4) `[W/M/L/C] Should show a unwanted software warning` - 3 - `SafetyNetJavaThreatType.UNWANTED_SOFTWARE`
- (6) `[A/W/M/L/C] Should show a billing warning` - 15 - `SafetyNetJavaThreatType.BILLING`

Resolves https://github.com/brave/brave-browser/issues/41581

This is a follow-up for https://github.com/brave/brave-core/pull/25842.

## Submitter Checklist:

- [x]  I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
There is no a strict test plan, because I never saw the website which had more than one threat reported by SafetyNet.

The basic sanity check though is
1. Open https://testsafebrowsing.appspot.com/
2. For 1, 2, 4, 6 option the corresponding interstitial page with warning must be displayed.

### CODE REVIEW

File: components/safe_browsing/android/java/src/org/chromium/components/safe_browsing/BraveSafeBrowsingApiHandler.java [hunk 1]
```diff
@@ -120,18 +123,18 @@ public void startUriLookup(long callbackId, String uri, int[] threatTypes, int p
...
                                         callbackId,
                                         LookupResult.SUCCESS,
                                         BraveSafeBrowsingUtils
                                                 .safetyNetToSafeBrowsingJavaThreatType(
-                                                        sbResponse
-                                                                .getDetectedThreats()
-                                                                .get(0)
-                                                                .getThreatType()),
+                                                        BraveSafeBrowsingUtils
+                                                                .getMostPriorityThreat(threats)),
```
Comment (fmarier) [inline components/safe_browsing/android/java/src/org/chromium/components/safe_browsing/BraveSafeBrowsingApiHandler.java:137]: nit:
```suggestion
                                                                .getHighestPriorityThreat(threats)),
```

File: components/safe_browsing/android/java/src/org/chromium/components/safe_browsing/BraveSafeBrowsingUtils.java [hunk 2]
```diff
@@ -139,4 +140,35 @@ public static int safetyNetToSafeBrowsingJavaThreatType(int safetyNetThreatType)
                 return SafeBrowsingJavaThreatType.NO_THREAT;
         }
     }
+
+    // Priorities from most high to low
+    private static final int SAFETY_NET_THREAT_PRIORITIES[] = {
+        SafetyNetJavaThreatType.SOCIAL_ENGINEERING,
+        SafetyNetJavaThreatType.BILLING,
+        SafetyNetJavaThreatType.POTENTIALLY_HARMFUL_APPLICATION,
```
Comment (fmarier) [inline components/safe_browsing/android/java/src/org/chromium/components/safe_browsing/BraveSafeBrowsingUtils.java:148]: I'd move POTENTIALLY_HARMFUL_APPLICATION to be second (i.e. between SOCIAL_ENGINEERING and BILLING).

File: components/safe_browsing/android/java/src/org/chromium/components/safe_browsing/BraveSafeBrowsingUtils.java [hunk 3]
```diff
@@ -139,4 +140,35 @@ public static int safetyNetToSafeBrowsingJavaThreatType(int safetyNetThreatType)
...
+        SafetyNetJavaThreatType.UNWANTED_SOFTWARE,
+        SafetyNetJavaThreatType.SUBRESOURCE_FILTER,
+        SafetyNetJavaThreatType.CSD_ALLOWLIST,
+        SafetyNetJavaThreatType.MAX_VALUE
+    };
+
+    @VisibleForTesting
+    public static int getMostPriorityThreat(List<Integer> threats) {
```
Comment (fmarier) [inline components/safe_browsing/android/java/src/org/chromium/components/safe_browsing/BraveSafeBrowsingUtils.java:156]:
```suggestion
    public static int getHighestPriorityThreat(List<Integer> threats) {
```

**Review by fmarier — APPROVED** [review]

File: components/safe_browsing/android/java/src/org/chromium/components/safe_browsing/BraveSafeBrowsingUtils.java (hunk 2 above)
Comment (AlexeyBarabash) [reply components/safe_browsing/android/java/src/org/chromium/components/safe_browsing/BraveSafeBrowsingUtils.java:147]: thanks, fixed at https://github.com/brave/brave-core/pull/26195/commits/cdb294ccfe8e78f31fe4404b4bd3264f885049a5

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Pull request merged**